src/components/Header.tsx

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { FormEvent, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { addTodo, updateTodo } from '../api/todos';
+
+interface Props {
+  todos: Todo[],
+  userId: number,
+  handleTempTodo: (todo: Todo | null) => void,
+  handleError: (error: string) => void;
+  handleIsUpdating: (status: boolean) => void,
+  handleUpdatingIds: (ids: number[]) => void,
+}
+
+export const Header: React.FC<Props> = ({
+  todos, userId, handleError, handleTempTodo,
+  handleIsUpdating, handleUpdatingIds,
+}) => {
+  const [todoTitle, setTodoTitle] = useState('');
+  const [isDisabled, setIsDisabled] = useState(false);
+  const hasTodos = todos.length > 0;
+
+  const handleCleaner = () => {
+    setIsDisabled(false);
+    setTodoTitle('');
+    handleIsUpdating(true);
+  };
+
+  const handleUpdateAllCompleted = () => {
+    handleIsUpdating(true);
+    let isAllCompleted = true;
+    const AllIds = todos.map(todo => {
+      if (!todo.completed) {
+        isAllCompleted = false;
+      }
+
+      return todo.id;
+    });
+
+    handleUpdatingIds(AllIds);
+    const updatedTodo = {
+      completed: !isAllCompleted,
+    };
+
+    AllIds.forEach(id => {
+      updateTodo(id, updatedTodo)
+        .then(() => handleCleaner())
+        .catch(() => handleError('Unable to update a todo'));
+    });
+  };
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+    const trimmedTitle = todoTitle.trim();
+
+    if (!trimmedTitle.length) {
+      handleError('Title can\'t be empty');
+
+      return;
+    }
+
+    setIsDisabled(true);
+    const newTodo: Todo = {
+      id: 0,
+      userId,
+      title: trimmedTitle,
+      completed: false,
+    };
+
+    handleTempTodo(newTodo);
+    addTodo(newTodo, userId)
+      .then(handleCleaner)
+      .catch(() => handleError('Unable to add a todo'));
+  };
```

Comment:
Create a separate function `addTodo` for this logic and wrap it to `try catch`.
Move this function to the app component